packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart

```diff
@@ -0,0 +1,632 @@
+  /// [WebViewAndroidWidget]. This comes at the cost of some performance on
+  /// Android versions below 10. See
+  /// https://flutter.dev/docs/development/platform-integration/platform-views#performance
+  /// for more information.
+  ///
+  /// Defaults to false.
+  final bool useHybridComposition;
+
+  /// Handles callbacks that are made by [android_webview.WebViewClient], [android_webview.DownloadListener], and [android_webview.WebChromeClient].
+  final WebViewPlatformCallbacksHandler callbacksHandler;
+
+  /// Manages named JavaScript channels and forwarding incoming messages on the correct channel.
+  final JavascriptChannelRegistry javascriptChannelRegistry;
+
+  /// Handles constructing [android_webview.WebView]s and calling static methods.
+  ///
+  /// This should only be changed for testing purposes.
+  final WebViewProxy webViewProxy;
+
+  /// Callback to build a widget once [android_webview.WebView] has been initialized.
+  final Widget Function(WebViewAndroidPlatformController controller)
+      onBuildWidget;
+
+  @override
+  State<StatefulWidget> createState() => _WebViewAndroidWidgetState();
+}
+
+class _WebViewAndroidWidgetState extends State<WebViewAndroidWidget> {
+  late final WebViewAndroidPlatformController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = WebViewAndroidPlatformController(
+      useHybridComposition: widget.useHybridComposition,
+      creationParams: widget.creationParams,
+      callbacksHandler: widget.callbacksHandler,
+      javascriptChannelRegistry: widget.javascriptChannelRegistry,
+      webViewProxy: widget.webViewProxy,
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    controller._release();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return widget.onBuildWidget(controller);
+  }
+}
+
+/// Implementation of [WebViewPlatformController] with the Android WebView api.
+class WebViewAndroidPlatformController extends WebViewPlatformController {
+  /// Construct a [WebViewAndroidPlatformController].
+  WebViewAndroidPlatformController({
+    required bool useHybridComposition,
+    required CreationParams creationParams,
+    required this.callbacksHandler,
+    required this.javascriptChannelRegistry,
+    @visibleForTesting this.webViewProxy = const WebViewProxy(),
+  })  : assert(creationParams.webSettings?.hasNavigationDelegate != null),
+        super(callbacksHandler) {
+    webView = webViewProxy.createWebView(
+      useHybridComposition: useHybridComposition,
+    );
+
+    webView.settings.setDomStorageEnabled(true);
+    webView.settings.setJavaScriptCanOpenWindowsAutomatically(true);
+    webView.settings.setSupportMultipleWindows(true);
+    webView.settings.setLoadWithOverviewMode(true);
+    webView.settings.setUseWideViewPort(true);
+    webView.settings.setDisplayZoomControls(false);
+    webView.settings.setBuiltInZoomControls(true);
```

Comment:
Do we know what the performance overhead of all of these being individual platform channel calls is?

This doesn't need to block proceeding, but it would be good to evaluate that to figure out if we should adjust the Dart/native boundary in a follow up for performance reasons (e.g., by adding a 'settings delta' struct that we could use to batch changes).

Reply:
So I did some testing with the with loading the `initialUrl` and the results I got were: (milliseconds)

```
WebView Widget (this implementation):
3016
2856
2882
2866
2925
3023
2803
2758
2934
2801

Avg: 2887.3

WebView Master (method channel implementation)
3702
3115
4877
3257
4581
3276
2998
2953
2789
2778

Avg excluding outliers: 3023
```

I haven't been able to find any noticeable differences between the implementations. But, I was also curious with the speed check when there were other `MethodCall`s being made while the `WebView` was loading so I added a constant `MethodCall` during the test:

```
Future<void> testCall() async {
    const MethodChannel('testChannel').invokeMethod<void>('');
    if (!stop) {
      await Future<void>.delayed(const Duration(milliseconds: 10));
      testCall();
    }
  }
```

And I received similar results.

The code for the test: 
```dart
testWidgets('initialUrl', (WidgetTester tester) async {
    final DateTime now = DateTime.now();
    final Completer<void> onPageFinished = Completer<void>();
    await tester.pumpWidget(
      MaterialApp(
        home: Directionality(
          textDirection: TextDirection.ltr,
          child: WebView(
            initialUrl: primaryUrl,
            onPageFinished: (_) {
              final int millis = DateTime.now().millisecondsSinceEpoch -
                  now.millisecondsSinceEpoch;
              print('$millis');
              onPageFinished.complete();
            },
          ),
        ),
      ),
    );
    await onPageFinished.future;
  });
```

Reply:
Awesome, good to know that overhead of crossing the boundary is minimal in general!